packages/nuxt/src/app/composables/fetch.ts

```diff
@@ -170,16 +170,28 @@ export function useFetch<
     }
 
     let _$fetch = opts.$fetch || globalThis.$fetch
+    const headers = Object.create(null)
 
     // Use fetch with request context and headers for server direct API calls
     if (import.meta.server && !opts.$fetch) {
       const isLocalFetch = typeof _request.value === 'string' && _request.value[0] === '/' && (!toValue(opts.baseURL) || toValue(opts.baseURL)![0] === '/')
       if (isLocalFetch) {
         _$fetch = useRequestFetch()
+        // Set default `accept` value to avoid it defaulting to `text/html` (from initial browser request)
+        headers.accept = 'application/json'
```

Comment:
Maybe instead avoid irritating accept? This seems a bug that we proxy browser's accept header down to API calls. ofetch defaults to the right accept header based on body but if we explicitly ovveride here, it cannot do this anymore.

Reply:
Okay, let's fix it upstream then 👍

Reply:
~> https://github.com/unjs/h3/pull/646